stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -639,20 +642,33 @@ public void onClose(Status status, Metadata trailers) {
      * Waits until there is a Runnable, then executes it and all queued Runnables after it.
      */
     public void waitAndDrain() throws InterruptedException {
-      Runnable runnable = queue.take();
-      while (runnable != null) {
+      Runnable runnable = poll();
+      if (runnable == null) {
+        waiter = Thread.currentThread();
+        try {
+          while ((runnable = poll()) == null) {
+            LockSupport.park(this);
```

Comment:
There is no synchronization, so the argument is misleading?

Reply:
@dapengzhang0 from [the docs](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/locks/LockSupport.html):
> This object is recorded while the thread is blocked to permit monitoring and diagnostic tools to identify the reasons that threads are blocked. ... The normal argument to supply as a `blocker` within a lock implementation is `this`.